### PR TITLE
updated to rc 16 and added annotations

### PR DESCRIPTION
**Pull request by J3fftw1**
## Short Description
Updated to rc 16 and added annotations

## Additions/Changes/Removals
Updated to rc 16 and added annotations

## Related Issues
N/A

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with base Slimefun and made sure nothing breaks/unexpected happens.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added which may not be obvious to maintainers.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

File: src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java [hunk 1]
```diff
@@ -60,12 +59,13 @@ public AdvancedSolarPanel(Type type) {
             });
     }
 
+
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java]: Remove the double empty line

File: src/main/java/dev/j3fftw/litexpansion/machine/MassFabricator.java [hunk 2]
```diff
@@ -125,11 +126,12 @@ private void tick(@Nonnull Block b) {
     }
 
     private boolean takePower(@Nonnull Block b) {
-        if (ChargableBlock.getCharge(b) < ENERGY_CONSUMPTION) return false;
-        ChargableBlock.addCharge(b, -ENERGY_CONSUMPTION);
+        if (getCharge((Location) b) < ENERGY_CONSUMPTION) return false;
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/machine/MassFabricator.java]: A Block isn't a Location, this will fail. #getLocation

**Review by WalshyDev — CHANGES_REQUESTED** [review]
Bunch of places where you cast Location reee

File: src/main/java/dev/j3fftw/litexpansion/machine/RubberSynthesizer.java [hunk 3]
```diff
@@ -122,11 +123,12 @@ private void tick(@Nonnull Block b) {
     }
 
     private boolean takePower(@Nonnull Block b) {
-        if (ChargableBlock.getCharge(b) < ENERGY_CONSUMPTION) return false;
-        ChargableBlock.addCharge(b, -ENERGY_CONSUMPTION);
+        if (getCharge((Location) b) < ENERGY_CONSUMPTION) return false;
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/machine/RubberSynthesizer.java]: same here then

**Review by WalshyDev — APPROVED** [review]

**Pull request merged**